conformance/lang/expressions/let-expr/let_expr.balt

```diff
@@ -0,0 +1,481 @@
+Test-Case: output
+Description: Test let-expr, when type-descriptor is a nil-type-descriptor which is a simple-type-descriptor.
+Labels: let-expr, module-class-defn, nil-literal, nil-type, return-stmt, union-type, boolean, record-type,
+        list-constructor-expr, mapping-constructor-expr, table-constructor-expr, query-expr
+        object-constructor-expr, new-expr, function-call-expr, method-call-expr, error-constructor-expr,
+        type-cast-expr, typeof-expr, unary-logical-expr, is-expr, equality-expr, logical-expr, conditional-expr
+
+class NilClass {
+    null n;
+    function init(null n) {
+        self.n = n;
+    }
+
+    function getNil() returns () {
+        return self.n;
+    }
+
+    function isNil(int? i) returns boolean {
+        return i == ();
+    }
+}
+
+type Record record {|
+    int|boolean? i;
+|};
+
+function init() {
+    io:println(() == let null n = null in n); // @output true
+    io:println(() == let () n = () in n); // @output true
```

Comment:
Please split up the null vs () tests also.

Reply:
Basically, if there are two separate labels, we can split them up into two tests. And then add an additional test, if necessary, combining them.

Reply:
Changed